great_expectations/checkpoint/types/checkpoint_result.py

```diff
@@ -284,97 +286,24 @@ def _list_validation_statistics(self) -> Dict[ValidationResultIdentifier, dict]:
-                    "runtime_parameters"
-                ].pop("batch_data")
-
-        json_dict = checkpointResultSchema.dump(self)
-        if len(batch_data_list) > 0:
-            for idx, val in enumerate(json_dict["checkpoint_config"]["validations"]):
-                if batch_data_list[idx] is not None:
-                    self.checkpoint_config["validations"][idx]["batch_request"][
-                        "runtime_parameters"
-                    ]["batch_data"] = json_dict["checkpoint_config"]["validations"][
-                        idx
-                    ][
-                        "batch_request"
-                    ][
-                        "runtime_parameters"
-                    ][
-                        "batch_data"
-                    ] = batch_data_list[
-                        idx
-                    ]
-        elif batch_data is not None:
-            self.checkpoint_config["batch_request"]["runtime_parameters"][
-                "batch_data"
-            ] = json_dict["checkpoint_config"]["batch_request"]["runtime_parameters"][
-                "batch_data"
-            ] = batch_data
-
-        return json_dict
+    def to_json_dict(self) -> dict:
+        return checkpointResultSchema.dump(self)
 
     def __repr__(self):
-        serializeable_dict = self.to_json_dict()
-        if len(serializeable_dict["checkpoint_config"].get("validations")) > 0:
-            for val in serializeable_dict["checkpoint_config"]["validations"]:
-                if (val["batch_request"].get("runtime_parameters") is not None) and (
-                    val["batch_request"]["runtime_parameters"].get("batch_data")
-                    is not None
-                ):
-                    val["batch_request"]["runtime_parameters"]["batch_data"] = str(
-                        type(val["batch_request"]["runtime_parameters"]["batch_data"])
-                    )
-
-        if serializeable_dict.get("batch_request") is not None:
-            if (
-                serializeable_dict["checkpoint_config"]["batch_request"].get(
-                    "runtime_parameters"
-                )
-                is not None
-            ) and (
-                serializeable_dict["checkpoint_config"]["batch_request"][
-                    "runtime_parameters"
-                ].get("batch_data")
-                is not None
-            ):
-                serializeable_dict["checkpoint_config"]["batch_request"][
-                    "runtime_parameters"
-                ]["batch_data"] = str(
-                    type(
-                        serializeable_dict["checkpoint_config"]["batch_request"][
-                            "runtime_parameters"
-                        ]["batch_data"]
-                    )
-                )
-
+        serializeable_dict: dict = self.to_json_dict()
+        batch_data_references: Tuple[
+            Optional[Any], Optional[List[Any]]
+        ] = get_runtime_parameters_batch_data_references_from_config(
```

Comment:
per the above comment re: these imports, rather than importing functions that work against config, consider instantiating a config object with the necessary methods / understanding of itself

Reply:
@donaldheppner These are static utility methods -- they will be moved to `batch.py` (in the next commit).